### PR TITLE
fix: missing EIP-2718 support checking tx type

### PR DESCRIPTION
**Pull request by vimpunk**
This PR fixes the test suite runner in light of new failing geth tests checking for whether on pre-Berlin forks non-legacy transactions are properly rejected. This was not the case for sputnik's evm-tests, because the Berlin hardfork implementation (https://github.com/rust-blockchain/evm/pull/40) opted to shift responsibility of implementing EIP-2718 (tx type filtering) to the layer hosting sputnik. In this case, the host is evm-test, which was missing that check.

**IMPORTANT**: This PR depends on https://github.com/rust-blockchain/evm/pull/162, which needs to be merged, and then the git submodule on this branch needs to be updated to master of rust-blockchain/evm.

### CODE REVIEW

**Comment by vimpunk** [comment]
@sorpaas, here's a new one for you :)

File: jsontests/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,8 @@ keywords = ["no_std", "ethereum"]
 edition = "2018"
 
 [dependencies]
-evm = { path = "../evm" }
+# evm = { path = "../evm" }
+evm = { git = "https://github.com/mandreyel/evm", branch = "feat/eip-4399" }
```
Comment (vimpunk) [inline jsontests/Cargo.toml]: TODO: Merge evm PR and then revert these changes.

Comment (sorpaas) [reply jsontests/Cargo.toml]: Please just revert this now. We only have CI in `evm` repo.

**Review by sorpaas — APPROVED** [review]

File: jsontests/Cargo.toml (hunk 1 above)
Comment (vimpunk) [reply jsontests/Cargo.toml]: Done, ready for merging.

**Comment by vimpunk** [comment]
@sorpaas ser, wen merge? :)

**Review by sorpaas — APPROVED** [review]

File: Cargo.lock [hunk 2]
```diff
@@ -581,6 +584,7 @@ dependencies = [
 [[package]]
 name = "evm-runtime"
 version = "0.37.0"
+source = "git+https://github.com/mandreyel/evm?branch=feat/eip-4399#d8324b41b5baf0a6f4221c3cacc3a1828a191e60"
```
Comment (sorpaas) [inline Cargo.lock]: Those should be reverted also.

Just run `cargo build` and commit the `Cargo.lock` file.

Comment (vimpunk) [reply Cargo.lock]: You're right, that was my bad. I reverted that and also added another fix for a sputnik fix I'm going to push soon. Please check the diff, it's small.

Comment (vimpunk) [reply Cargo.lock]: Also, while I have your attention, can we merge https://github.com/rust-blockchain/evm/pull/162? It's been sitting open for a while but it's done.

**Review by sorpaas — APPROVED** [review]
